python/az/aro/azext_aro/custom.py

```diff
@@ -191,13 +207,67 @@ def aro_list_credentials(client, resource_group_name, resource_name):
     return client.list_credentials(resource_group_name, resource_name)
 
 
-def aro_update(client, resource_group_name, resource_name, no_wait=False):
-    oc = v2020_04_30.OpenShiftClusterUpdate()
+def aro_update(cmd, client, resource_group_name, resource_name, no_wait=False):
+    try:
+        oc = client.get(resource_group_name, resource_name)
 
-    return sdk_no_wait(no_wait, client.update,
-                       resource_group_name=resource_group_name,
-                       resource_name=resource_name,
-                       parameters=oc)
+        resources = get_ocp_cluster_network_resources(cmd.cli_ctx, oc)
+
+        aad = AADManager(cmd.cli_ctx)
+        rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
+        rp_client_sp = None
+
+        try:
+            rp_client_sp = aad.get_service_principal(rp_client_id)
+        except GraphErrorException:
+            pass
+
+        if not rp_client_sp:
+            logger.warning("The rp service principal does not exist. Cannot recreate.")
```

Comment:
@jim-minter  we need to confirm these error messages for the end users. What message we want to give them